### PR TITLE
[SD-2795] Failure to re-login in case of LDAP authentication.

### PR DESCRIPTION
**Pull request by mdhaman**

### CODE REVIEW

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

**Comment by mugurrus** [comment]
:+1:

File: server/superdesk/tests.py [hunk 1]
```diff
@@ -146,12 +146,16 @@ def get_prefixed_url(current_app, endpoint):
...
+    :param context: test context
+    :param dict user: user
+    """
     user = user or test_user
     with context.app.test_request_context(context.app.config['URL_PREFIX']):
         original_password = user['password']
 
-        if user.get('user_type') is None:
-            user['user_type'] = 'administrator'
+        user.setdefault('user_type', 'administrator')
```
Comment (plamut) [inline server/superdesk/tests.py]: In general I like this change, very Pythonic. However, it might not exactly equivalent to the old code - can it happen that the `user_type` key exists but is set to `None`? If yes, then the user will not be set to administrator anymore as it was in the old version.

File: server/superdesk/tests.py [hunk 2]
```diff
@@ -176,10 +186,26 @@ def setup_ad_user(context, user):
     ad_user = ad_user.copy()
     ad_user['email'] = 'mock@mail.com.au'
 
-    if ad_user.get('user_type') is None:
-        ad_user['user_type'] = 'administrator'
+    ad_user.setdefault('user_type', 'administrator')
```
Comment (plamut) [inline server/superdesk/tests.py]: Same remark as above, the old and new code are not exactly the same.

**Comment by plamut** [comment]
Looks good in general, but please double check that we haven't accidentally introduced a bug here.

File: server/superdesk/tests.py (hunk 1 above)
Comment (mdhaman) [inline server/superdesk/tests.py]: `user_type` is not provided. There are some test where `user_type` is provided via the step definition. It should be fine. merging the code for now as it is blocker for our ldap configuration.

Comment (plamut) [inline server/superdesk/tests.py]: Perfectly fine with me, you probably know the possible edge cases better than me. :wink:
